internal/dcache/clustermap/clustermap.go

```diff
@@ -248,42 +248,22 @@ func ReportRVOffline(rvName string) error {
 	return nil
 }
 
-// Tell clustermanager to update the state of a component RV for an MV.
 func UpdateComponentRVState(mvName string, rvName string, rvNewState dcache.StateEnum) error {
-	// Clustermanager must call RegisterComponentRVStateUpdater() in startup, so we don't expect this to be nil.
-	common.Assert(componentRVStateUpdater != nil)
-
-	//
-	// When multiple threads call UpdateComponentRVState() it's not useful to let all of them proceed
-	// and fight it out using optimistic concurrency. That results in far more retries than if we coordinate
-	// and only allow one thread to update the global clustermap at one time. This happens when one or more
-	// RVs go down and we start one sync job for each replica, with 100s of MVs, there are 100s of sync jobs
-	// which run syncComponentRV() and call UpdateComponentRVState() multiple times.
-	//
-	// TODO: This still results in as many GetBlob/PutBlob calls as the number of threads calling this
-	//       function. We should try to batch the updates that come in close time proximity.
-	//       We can have a channel where we add the required updates and one go routine can then update the
-	//       global clustermap once every few seconds.
-	//
-	updateLock.Lock()
-	defer updateLock.Unlock()
-
-	return componentRVStateUpdater(mvName, rvName, rvNewState)
+	updateComponentRVStateChannel <- dcache.ComponentRVUpdateMessage{MvName: mvName, RvName: rvName, RvNewState: rvNewState}
```

Comment:
UpdateComponentRVState() cannot return till the requested change is saved in the clustermap.
The batching change should not change the semantics.
Multiple threads calling UpdateComponentRVState() with disjoint changes will have their changes queued to a list and then that  list will be periodically collated and published into clustermap, but only after a given change is published can the corresponding UpdateComponentRVState() can return.